tests/models/demultiplexing/test_flowcell_model.py

```diff
@@ -134,3 +135,52 @@ def test_get_sample_model_novaseq_x(novaseq_x_flow_cell: FlowCellDirectoryData):
 
     # THEN it is FlowCellSampleNovaSeq6000Bcl2Fastq
     assert sample_model == FlowCellSampleNovaSeqX
+
+
+def test_get_bcl_converter_by_sequencer(
+    flow_cell_directory_name_demultiplexed_with_bcl2fastq: str,
+):
+    """Test that the bcl converter of a bcl2fastq flow cell is bcl2fastq."""
+    # GIVEN a Bcl2Fastq flow cell directory
+
+    # WHEN instantiating a flow cell object
+    flow_cell = FlowCellDirectoryData(
+        flow_cell_path=Path(flow_cell_directory_name_demultiplexed_with_bcl2fastq)
+    )
+
+    # THEN it sets the converter to blc2fastq
+    assert flow_cell.bcl_converter == BclConverter.BCL2FASTQ
+
+
+def test_flow_cell_directory_data_with_set_bcl_converter(
+    flow_cell_directory_name_demultiplexed_with_bcl2fastq: str, bcl_converter=BclConverter.DRAGEN
+):
+    """Test that the bcl converter is set to the specified value."""
+    # GIVEN a Bcl2Fastq flow cell directory
+
+    # GIVEN the bcl converter is set to dragen
+
+    # WHEN instantiating a flow cell object
+    flow_cell = FlowCellDirectoryData(
+        flow_cell_path=Path(flow_cell_directory_name_demultiplexed_with_bcl2fastq),
+        bcl_converter=bcl_converter,
+    )
+
+    # THEN the bcl converter is dragen
+    assert flow_cell.bcl_converter == bcl_converter
+
+
+def test_flow_cell_directory_data_with_novaseq_flow_cell_directory(
+    flow_cell_directory_name_demultiplexed_with_bcl_convert: str, bcl_converter=BclConverter.DRAGEN
+):
+    """Test that the bcl converter is set to dragen when prodiving a novaseq flow cell directory."""
+    # GIVEN a Bcl2Fastq flow cell directory
+
+    # WHEN instantiating a flow cell object
+    flow_cell = FlowCellDirectoryData(
+        flow_cell_path=Path(flow_cell_directory_name_demultiplexed_with_bcl_convert),
+        bcl_converter=bcl_converter,
```

Comment:
I interpret this test as wanting to show that if the flow cell was demultiplexed with BCL convert, BCL convert is set automatically from the flow cell dir name? If so, should we not omit setting the bcl_converter to BclConverter.DRAGEN here?

Reply:
agree